### PR TITLE
HIVE-24949: Fail to rename a partition with customized catalog

### PR DESCRIPTION
**Pull request by dengzhhu653**

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java [hunk 1]
```diff
@@ -600,7 +599,7 @@ public Partition alterPartition(RawStore msdb, Warehouse wh, String catName, Str
     Database db;
     try {
       msdb.openTransaction();
-      Table tbl = msdb.getTable(DEFAULT_CATALOG_NAME, dbname, name, null);
```
Comment (kgyrtkirk) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java:603]: this also seems like a bugfix :)

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java [hunk 2]
```diff
@@ -9768,7 +9767,8 @@ private void writeMPartitionColumnStatistics(Table table, Partition partition,
 
       for (ColumnStatisticsObj statsObj : statsObjs) {
         MPartitionColumnStatistics mStatsObj =
-            StatObjectConverter.convertToMPartitionColumnStatistics(mPartition, statsDesc, statsObj, colStats.getEngine());
+            StatObjectConverter.convertToMPartitionColumnStatistics(mPartition, statsDesc, statsObj,
+                colStats.getEngine(), getDefaultCatalog(conf));
```
Comment (kgyrtkirk) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java:9771]: I think the catalog name should be already accessible thru: `MPartition -> MTable -> MDatabase ->catalogName`

Comment (dengzhhu653) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java]: you are right, fix as suggested, the `MPartition` is nullable here due to https://github.com/apache/hive/blob/baf8247af78f35a68fab0756f9d571c73cdf5f96/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java#L188-L192, fill the statsDesc with the catalog in this case: https://github.com/apache/hive/pull/2910/files#diff-f832bc7a00499f47169832b5ab5339dc879c994bfb955bd0c4b3ece6c236d9dcR186-R188
Thank you for the review!

Comment (dengzhhu653) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java]: Hi @kgyrtkirk, could you please take another look if have some secs?

**Review by kgyrtkirk — APPROVED** [review]

**Pull request merged**